nofos/nofos/forms.py

```diff
@@ -33,15 +33,47 @@ def __init__(self, *args, **kwargs):
 create_nofo_form_class = create_object_model_form(Nofo)
 
 
-# Creating form classes
-NofoNameForm = create_nofo_form_class(
-    ["title", "short_name"], not_required_labels=["Short name"]
-)
+# Creating form classes with validation
+class NofoNameForm(forms.ModelForm):
+    class Meta:
+        model = Nofo
+        fields = ["title", "short_name"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["short_name"].required = False
+
+    def clean_title(self):
+        title = self.cleaned_data.get("title")
+        if title is None or not title.strip():
+            raise forms.ValidationError("Title cannot be empty.")
+        return title.strip()
```

Comment:
Do we need these extra forms? I think this logic is already applied in our `create_nofo_form_class`. 

There is a better validation message from these forms though.

Were you able to find a scenario in the previous forms that were letting bad values through?

Reply:
My concern is that we have a factory function which is creating almost all of our attribute-level forms, and I don't really want to fork that into other base classes if we don't need to (particularly where seems like we aren't relying on extra functionality).